### PR TITLE
Adding status of rule to output. 

### LINKED ISSUE #312: wsk rule list should show whether a rule is enabled or not
wsk rule list should show whether a rule is enabled or not

### PR DESCRIPTION
**Pull request by jessealva**
The output looks like this now

```
rules
/guest/rule-hello2                                                     private              active
/guest/rule-hello                                                       private              inactive
```
 
Fixes #312

Still working on test cases for this, just creating the pull request for feedback

### CODE REVIEW

**Comment by jessealva** [comment]
@dubeejw Looking for feedback

**Comment by dubeejw** [comment]
Looks like the formatting is off for long entity names. There is a related issue here: https://github.com/apache/incubator-openwhisk/issues/2484.
```
/namespace/testRule                                       private              active
/namespace/r                                              private              active
/namespace/On Triggers non-feed triggers --using kkkkkkkkkkkkk --invoke invoke4 private              active
/namespace/invoke-periodically                            private              active
/namespace/rra                                            private              inactive
/namespace/myRule                                         private              active
/namespace/myRule2                                        private              active
/namespace/thumbnail_cloudant_rule519                     private              active
/namespace/kjjhjh                                         private              inactive
/namespace/On alarms alarm --using oioijoijoijoj --invoke Hello World With Params private              active
/namespace/On github webhook --using uiTest --invoke github private              active
```

File: tools/cli/go-whisk-cli/commands/rule.go [hunk 1]
```diff
@@ -388,6 +388,18 @@ var ruleListCmd = &cobra.Command{
...
             return werr
+        } else {
+            //No errors, lets attempt to retrieve the status of each rule #312
+            for index, rule := range rules {
+                ruleStatus, _, err := client.Rules.Get(rule.Name)
+                if err != nil {
+                    rules[index].Status = wski18n.T("Unknown")
+                    errStr := wski18n.T("Unable to get status of rule '{{.name}}': {{.err}}",
+                        map[string]interface{}{"name": rule.Name, "err": err})
+                    fmt.Println(errStr)
```
Comment (dubeejw) [inline tools/cli/go-whisk-cli/commands/rule.go]: Should return the error after the print statement.

File: tests/src/test/scala/system/basic/WskRuleTests.scala [hunk 2]
```diff
@@ -371,4 +371,24 @@ class WskRuleTests
...
+
+            ruleSetup(Seq(
+                (ruleName, triggerName, (actionName, actionName, defaultAction)),
+                (ruleName2, triggerName, (actionName, actionName, defaultAction))),
+                assetHelper)
+
+            wsk.rule.disable(ruleName)
+            wsk.rule.list().stdout should include(ruleName)
+            wsk.rule.list().stdout should include(ruleName2)
+            wsk.rule.list().stdout should include("active")
```
Comment (dubeejw) [inline tests/src/test/scala/system/basic/WskRuleTests.scala]: Make sure the active and inactive statuses are connected to the correct rule names

File: tools/cli/go-whisk-cli/commands/rule.go [hunk 3]
```diff
@@ -388,6 +388,18 @@ var ruleListCmd = &cobra.Command{
                     map[string]interface{}{"name": getClientNamespace(), "err": err})
             werr := whisk.MakeWskErrorFromWskError(errors.New(errStr), err, whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG, whisk.NO_DISPLAY_USAGE)
             return werr
+        } else {
+            //No errors, lets attempt to retrieve the status of each rule #312
+            for index, rule := range rules {
+                ruleStatus, _, err := client.Rules.Get(rule.Name)
+                if err != nil {
+                    rules[index].Status = wski18n.T("Unknown")
```
Comment (dubeejw) [inline tools/cli/go-whisk-cli/commands/rule.go]: Shouldn't need this. Just error out.

**Comment by jessealva** [comment]
@dubeejw is there a specific way that you would like the long names to be truncated? Has that already been discussed?

**Comment by dubeejw** [comment]
@jessealva, we can address the name truncations in a different PR. I opened a separate issue for it here: https://github.com/apache/incubator-openwhisk/issues/2484.

**Comment by jessealva** [comment]
So fixed all commented portions except the long name problem, but thats being taken care of in #2484 .  

https://openwhisk-jenkins.swg-devops.com:8443/view/Playgrounds/job/Playground1/1854/

@dubeejw ^^

**Comment by jessealva** [comment]
PG1 #1854

**Comment by dubeejw** [comment]
@jessealva, looks like you have a lot of commits here from upstream master that shouldn't be listed.

**Comment by rabbah** [comment]
```
git fetch --all --prune
git rebase upstream/master
```

is your friend :)
avoid merge commits and check out the git guide for more tips. https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines

**Comment by jessealva** [comment]
Fixed, @dubeejw 
PG 1, 1857

File: tests/src/test/scala/system/basic/WskRuleTests.scala [hunk 4]
```diff
@@ -371,4 +370,23 @@ class WskRuleTests
...
+            val triggerName = withTimestamp("ruleDisableTrigger")
+            val actionName = withTimestamp("ruleDisableAction")
+
+            ruleSetup(Seq(
+                (ruleName, triggerName, (actionName, actionName, defaultAction)),
+                (ruleName2, triggerName, (actionName, actionName, defaultAction))),
+                assetHelper)
+
+            wsk.rule.disable(ruleName)
+            val listOutput = wsk.rule.list().toString().split("\n")
```
Comment (rabbah) [inline tests/src/test/scala/system/basic/WskRuleTests.scala]: tip: you can replace `.split("\n")` with `.lines`

Comment (markusthoemmes) [inline tests/src/test/scala/system/basic/WskRuleTests.scala]: Also, shouldn't it be `wsk.rule.list().stdout.lines`?

**Comment by jessealva** [comment]
Haha, that last update had me scratching my head a bit, until i realized i had an iterator and my second check was looking for text the iterator had already past when doing the first check.  Changed the checks to so they are done in the order in which the strings they are looking for are printed and everything was fine again.

**Comment by dubeejw** [comment]
@csantanapr, want to merge?

**Comment by jessealva** [comment]
PG1 #1871

**Comment by rabbah** [comment]
i can merge it - @jessealva be careful using hashtag referring to pgs; github will incorrectly cross link other issues. I like pgN/YYY.

**Pull request merged**